client/client-multi/client-objective-c/Kaa/channel/impl/channels/DefaultOperationTcpChannel.m

```diff
@@ -281,9 +281,11 @@ - (void)stream:(NSStream *)aStream handleEvent:(NSStreamEvent)eventCode {
             __weak typeof(self) weakSelf = self;
             [self.executor addOperationWithBlock:^{
                 [weakSelf sendConnect];
-                [weakSelf schedulePingTask];
             }];
-            
+            dispatch_time_t time =  dispatch_time(DISPATCH_TIME_NOW, (int64_t)PING_TIMEOUT_SEC * NSEC_PER_SEC);
+            dispatch_after(time, dispatch_get_global_queue(QOS_CLASS_UTILITY, 0), ^{
```

Comment:
Produces another issue: blocks global queue. Instead of that you may try to schedule with delay the same operations that are done inside PingTask but it has another weak place: no ability to cancel ping task